core/src/main/java/org/opensearch/sql/planner/physical/PhysicalPlanNodeVisitor.java

```diff
@@ -79,4 +79,8 @@ public R visitMLCommons(PhysicalPlan node, C context) {
   public R visitAD(PhysicalPlan node, C context) {
     return visitNode(node, context);
   }
+
+  public R visitHighlight(PhysicalPlan node, C context) {
```

Comment:
What's the difference between visitHighlight and visitAD?

Reply:
AD and highlight operators need their own visiting functions to be called from their derived classes. You can find the overriding functions in `OpenSearchExecutionProtector`. The nodes passed into these functions will be for their respective `HighlightOperator` and `ADOperator`. This way the operator nodes are passed in, we can successfully visit the input for each node.